Support for Expo SDK 54 and new architecture.

## Summary

React-native-get-random-values is breaking builds with Expo SDK 54 + New Architecture.
Technical Details:
- Error: Undefined symbols for architecture arm64: _RCTRegisterModule
- Root Cause: react-native-get-random-values@1.11.0 (used by Segment packages) has New Architecture compatibility issues
- Build Failure: iOS builds fail during linking phase because the module can't find React Native registration symbols

Why This Matters:
  - react-native-get-random-values is only maintained by one person (LinusU)
  - Very little activity on the GitHub repository
  - Blocking Expo SDK 54 adoption for any app using Segment Analytics
  - Unreliable dependency for enterprise/production apps

  Current Segment Dependencies:
  - @segment/analytics-react-native@2.21.3 → depends on react-native-get-random-values@1.11.0
  - @segment/sovran-react-native@1.1.3 → depends on react-native-get-random-values@1.11.0

## Proposed Solution
  Recommended Solution for Segment:
  Replace react-native-get-random-values

--- a/package.json
+++ b/package.json
@@ -52,6 +52,7 @@
     "react": "18.2.0",
     "react-native": "^0.72.7",
     "react-native-builder-bob": "^0.23.1",
+    "react-native-get-random-values": "^1.11.0",
     "rimraf": "^5.0.5",
     "semantic-release": "^22.0.12",
     "semantic-release-yarn": "^3.0.2",

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -63,7 +63,7 @@
     "@segment/sovran-react-native": "^1.1.0",
     "react": "*",
     "react-native": "*",
-    "react-native-get-random-values": "1.x"
+    "react-native-get-random-values": "1.x || 2.x"
   },
   "peerDependenciesMeta": {
     "@react-native-async-storage/async-storage": {

--- a/packages/sovran/package.json
+++ b/packages/sovran/package.json
@@ -68,7 +68,8 @@
   "peerDependencies": {
     "@react-native-async-storage/async-storage": "2.x",
     "react": "*",
-    "react-native": "*"
+    "react-native": "*",
+    "react-native-get-random-values": "1.x || 2.x"
   },
   "peerDependenciesMeta": {
     "@react-native-async-storage/async-storage": {
@@ -78,7 +79,6 @@
   "dependencies": {
     "ansi-regex": "5.0.1",
     "deepmerge": "^4.2.2",
-    "react-native-get-random-values": "1.x",
     "shell-quote": "1.8.0",
     "uuid": "^14.0.0"
   },

